From gnutls-devel-bounces+xxxxxxxx=yyyyyyyyyyyyyyyyyy@gnu.org Sun Sep  2 15:34:24 2012
Message-ID: <50436094.5000900@gnutls.org>
Date: Sun, 02 Sep 2012 15:35:16 +0200
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
MIME-Version: 1.0
To: gnutls-devel@gnu.org
Subject: Re: gnutls 3.1.0 spurious test-suite failure
Content-Type: multipart/mixed; boundary="------------030701000602060700030807"
List-Id: GnuTLS development discussions <gnutls-devel.gnu.org>

This is a multi-part message in MIME format.
--------------030701000602060700030807
Content-Type: text/plain; charset=ISO-8859-1
Content-Transfer-Encoding: 7bit

On 09/02/2012 02:48 PM, Andreas Metzler wrote:

> On 2012-09-02 Nikos Mavrogiannopoulos <nmav@gnutls.org> wrote:
> [...]
>> Hello Andreas,
>>  It seems like a strange error. select() fails for some reason in these
>> systems in client side. Could you apply the two patches below (the
>> difference is that the print the errno) and re-run the test with -v?
>> This would provide an idea of why select fails there.
> 
> There you go:
> https://buildd.debian.org/status/fetch.php?pkg=gnutls28&arch=amd64&ver=3.1.0-4&stamp=1346588778
> https://buildd.debian.org/status/package.php?p=gnutls28&suite=experimental


Does the attached patch fix the issue?

regards,
Nikos


--------------030701000602060700030807
Content-Type: text/plain;
 name="patch.txt"
Content-Transfer-Encoding: 7bit
Content-Disposition: attachment;
 filename="patch.txt"

diff --git a/lib/gnutls_buffers.c b/lib/gnutls_buffers.c
index ff0f82a..e03af48 100644
--- a/lib/gnutls_buffers.c
+++ b/lib/gnutls_buffers.c
@@ -661,10 +661,9 @@ _gnutls_io_check_recv (gnutls_session_t session, unsigned int ms)
   reset_errno (session);
 
   ret = session->internals.pull_timeout_func(fd, ms);
-
-  err = get_errno (session);
   if (ret == -1)
     {
+      err = get_errno (session);
       _gnutls_read_log ("READ_TIMEOUT: %d returned from %p, errno=%d (timeout: %u)\n",
 			(int) ret, fd, err, ms);
       return errno_to_gerr(err);
diff --git a/lib/system.c b/lib/system.c
index 14bd085..067f189 100644
--- a/lib/system.c
+++ b/lib/system.c
@@ -125,7 +125,7 @@ int system_recv_timeout(gnutls_transport_ptr_t ptr, unsigned int ms)
 {
 fd_set rfds;
 struct timeval tv;
-int ret, ret2;
+int ret;
 int fd = GNUTLS_POINTER_TO_INT(ptr);
 
   FD_ZERO(&rfds);
@@ -144,10 +144,6 @@ int fd = GNUTLS_POINTER_TO_INT(ptr);
   if (ret <= 0)
     return ret;
 
-  ret2 = recv(fd, NULL, 0, MSG_PEEK);
-  if (ret2 == -1)
-    return ret2;
-      
   return ret;
 }
 

--------------030701000602060700030807
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Disposition: inline

_______________________________________________
Gnutls-devel mailing list
Gnutls-devel@gnu.org
https://lists.gnu.org/mailman/listinfo/gnutls-devel

--------------030701000602060700030807--


